Feature/90 housing available base data structure

VitaRox opened the pull request:
In short, this defines the schema representing the object which represents the rental listing in the HousingAvailable section of the app;

VitaRox committed:
- Initial commit: adds housing-available directory and housingAvailable.models.js, which
	will outline the basic data structure of our rental listings;

VitaRox committed:
- Fleshes out housingAvailable.modles.js in the following ways:
-- Defines characteristics of both the rental properties themselves and the landlords who seek to rent them;
-- Defines relationships between different schema (HousingAvailable, Landlord, Property);
-- Connects this schema's data to our api;
-- Defines which fields (columns/properties) may be left blank or null;
-- Defines the HousingAvailable object itself which is represented by the HousingAvailable schema;
-- Maps its very own ApiFilterQueries to the relevant apiFilterQuery() of each object comprising our schema
	(Property & Landlord);

VitaRox committed:
- Adds 'body' to ApiPatch;
- Adds 'status' property to schema, object, apiFilterQuery, apiPatch, which
	tells what status the property is in vis-a-vis the rental process;
- In last commit: I forgot to mention that the last commit also adds postgres-data to my .gitignore file;

zaclittleberry commented:
# All
status should just be a Sequelize.STRING. enum or something else isn't necessarily a bad choice, but we don't know what those values will be and string will be more flexible for us.

the module you added should be added to lib/modules.js. You might get compilation errors after that because now your code is actually being run and it has problems. This is the right path to take, and work through those errors.

*.routes.js and *.controllers.js should have the outer structure of those files in them. basically, the module.exports statement, and the bare structure to get to their return statments. the returns can just be [] and {} for routes and controllers respectively.

AuthorId and location will be added via association. they don't get defined as fields on the 'main' part of the model.

# Vita

your editor removed the newline at the end of .gitignore. git actually prefers the newline. if you could configure your editor to automatically add it/leave it, that'd be good.

please make your file name housing-available.models.js. JS uses camelCase for variable names, but our preference is dash-style for file names.

please make contactName to just be contact (to follow design pattern of property), and it will replace need for phone and email and preferred contact method so those can be removed (contact is kind of treated as a fallback to being attached to a landlord, which has 'proper' contact fields).

Author association should stay attached to Account, not Landlord.

You're onto something with Landlord though. We should add that as an association: HousingAvailable belongsTo Landlord, no need for the second param (the 'as' statement).

The association to Property should be a belongsTo, and no need for the config object w/ as.

details should just be set to a JSONB field (just as metadata is done). jsonb data structure in postgres is flexible, so we don't define the sub-fields in the database. We'll handle it in input validation and controller logic, but don't worry about that for now (you can save this to a file for reference later).

*.models.js only need the db property for this ticket. we'll add models for the routes as we build out routes, so please remove those for now (it might be useful to save them in a textfile on your computer for later since you already put the work into creating them. They may end up looking different, but at least you'll still have something to reference.)

zaclittleberry reviewed (changes requested):
see comments in conversation tab

VitaRox committed:
- Initial commit;
- This branch seeks to remedy the merge issues id'ed in pull request;

VitaRox committed:
- Adds in renamed 'housing-available' directory;

VitaRox committed:
- Adds in skeletal 'routes' and 'controllers' modules to our 'housing-available' directory;
- Removes 'AuthorId' and 'location' from housing-available.models.js;
- Adds export statement to our lib/modules.js module;
- Remove explicit sub-fields (properties) of 'details' field;

VitaRox committed:
- Adds in our skeletal controllers and routes modules, as the last commit apparently didn't accomplish this;

VitaRox committed:
- Adds in a blank test file so passes gitlab pipeline test;

VitaRox committed:
- Fixes gitlab pipeline error due to misnamed 'tests' file in 'housing-available'; file was named 'test' instead;

VitaRox committed:
- Adds in belongsTo Association to Account and Location;
- Adds project config (.vs) file to .gitignore;

zaclittleberry committed:
Merge branch 'story/housing-wanted-available' into feature/90-housing-available-base-data-structure

Commit:
Merge branch 'feature/90-housing-available-base-data-structure' into 90-housing-available-base-data-structure/revision_001

Commit:
Merge pull request #80 from GazeDev/90-housing-available-base-data-structure/revision_001

90 housing available base data structure/revision 001

Commit:
Update housing-available.models.js
